roles/compute_resources/README.md

```diff
@@ -0,0 +1,44 @@
+theforeman.foreman.compute_resources
+====================================
+
+This role creates and manages Compute Resources.
+
+Role Variables
+--------------
+
+This role supports the [Common Role Variables](https://github.com/theforeman/foreman-ansible-modules/blob/develop/README.md#common-role-variables).
+
+The main data structure for this role is the list of `foreman_compute_resources`. Each `compute_resource` requires the following fields:
+
+- `name`: The name of the compute resource.
+
+The following fields are optional and will be omitted by default:
+
+- `description`: Description of the compute resource
+- `provider`: Compute resource provider. Required if *state=present_with_defaults*.
+- `provider_params`: Parameter specific to compute resource provider. Required if *state=present_with_defaults*.
+- `updated_name`: New compute resource name. Setting this will make the role not idempotent.
+
+Example Playbooks
+-----------------
+
+Create a lifecycle environment path with three environments: Library -> Dev -> Test -> Prod
+
+```yaml
+- hosts: localhost
+  roles:
+    - role: theforeman.foreman.compute_resources
+      vars:
+        foreman_server_url: https://foreman.example.com
+        foreman_username: "admin"
+        foreman_password: "changeme"
+        foreman_organization: "Default Organization"
+        foreman_compute_resources:
+          - name: "VMware"
+            provider: "vmware"
+            provider_params:
+              url: "vcenter.example.com"
+              user: "administrator@vsphere.local"
+              password: "changeme"
+              datacenter: "ha-datacenter"
```

Comment:
I know, we talked about (operating-system/compute-resource -) images. Would it be a good architecture to define the images as a list in this Compute-resource dict?

Reply:
I definitely see some logical reasons for this since images are tightly coupled with compute resources (must have the latter to do the former). Should I add it?

Reply:
I'm not in the position to decide, but i'd rather say yes.
cc @ehelms

Reply:
I agree with adding it.

Reply:
Done.